osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSlider.cs

```diff
@@ -179,6 +186,8 @@ protected override void ClearNestedHitObjects()
             tailContainer.Clear(false);
             repeatContainer.Clear(false);
             tickContainer.Clear(false);
+
+            OverlayElementContainer.Clear();
```

Comment:
How do things get re-proxied? It would require components to get re-`LoadComplete`'d.

Reply:
Is that why they are disappearing? I thought this was called on skin change, but I guess it's also happening on pooling?

Reply:
Right. Ignoring that, does this general direction feel better than the other PR?

Reply:
Yeah it looks okay.

Reply:
I've gone ahead and applied a fix for the parts not reproxying (79438c1). It's a little verbose but I don't think it's *horrible*.

The main gotchas are:

* drawables need to keep their own proxy around because you cannot unproxy
* nested drawable objects cannot reference their parent drawable objects willy nilly as they are not coupled due to being pooled separately, so about the only time where the reproxying can happen is on hitobject application